pkgs/applications/editors/vscode/with-extensions.nix

```diff
@@ -70,6 +70,6 @@ runCommand "${wrappedPkgName}-with-extensions-${wrappedPkgVersion}" {
   ln -sT "${vscode}/share/applications/${executableName}.desktop" "$out/share/applications/${executableName}.desktop"
   ln -sT "${vscode}/share/applications/${executableName}-url-handler.desktop" "$out/share/applications/${executableName}-url-handler.desktop"
   makeWrapper "${vscode}/bin/${executableName}" "$out/bin/${executableName}" ${lib.optionalString (vscodeExtensions != []) ''
-    --add-flags "--extensions-dir ${combinedExtensionsDrv}/share/${wrappedPkgName}/extensions"
+    --add-flags "--extensions-dir ${combinedExtensionsDrv}/share/vscode/extensions"
```

Comment:
If you're going to pin the directory as `$out/share/vscode`, you'll have to pin it at the `vscodeExtension` as well, else this would break insiders which are installed in `vscode-insiders`.

Reply:
Thanks, I wasn't aware of `vscode-insiders`.

Does this work at all though? I tried running it with the following `shell.nix`:

```
{ pkgs ? import <nixpkgs> { config.allowUnfree = true; } }:
pkgs.mkShell {
  buildInputs = [
    (pkgs.vscode.override {
      isInsiders = true;
    })
  ];
}
```

but got:

```
$ nix-shell
these derivations will be built:
  /nix/store/0nz6hqdkkmnnj4rv5872w7wpl6dfmyq8-code-insiders.desktop.drv
  /nix/store/3bm496wq8q5wy50yi106swj7yj69bk0n-code-insiders-url-handler.desktop.drv
  /nix/store/3s1w1xh5yihnirwdggk0ypip67bcfcd5-vscode-1.41.0.drv
building '/nix/store/3bm496wq8q5wy50yi106swj7yj69bk0n-code-insiders-url-handler.desktop.drv'...
building '/nix/store/0nz6hqdkkmnnj4rv5872w7wpl6dfmyq8-code-insiders.desktop.drv'...
building '/nix/store/3s1w1xh5yihnirwdggk0ypip67bcfcd5-vscode-1.41.0.drv'...
unpacking sources
unpacking source archive /nix/store/gyps4ls3zrfs1b9rjbln4aq19k7hlll4-VSCode_1.41.0_linux-x64.tar.gz
source root is VSCode-linux-x64
setting SOURCE_DATE_EPOCH to timestamp 1576089152 of file VSCode-linux-x64/resources/app/node_modules.asar
patching sources
glibPreInstallPhase
installing
substitute(): ERROR: file '/nix/store/p3hrd9k8hq3k134ma7vnlw98hj6lg0ma-vscode-1.41.0/lib/vscode/bin/code-insiders' does not exist
builder for '/nix/store/3s1w1xh5yihnirwdggk0ypip67bcfcd5-vscode-1.41.0.drv' failed with exit code 1
error: build of '/nix/store/3s1w1xh5yihnirwdggk0ypip67bcfcd5-vscode-1.41.0.drv' failed
$
```

(this is on nixos-unstable b0bbacb52134a7e731e549f4c0a7a2a39ca6b481)

Reply:
The source is pinned on a stable version, if someone wants to use the insiders version, they need to change the `src` as well.

```nix
  vscode = (
    super.vscode.override {
      isInsiders = true;
    }
  ).overrideAttrs (
    _: rec {
      name = "${pname}-${version}";
      pname = "vscode-insiders";
      version = "nightly";

      src = builtins.fetchurl {
        name = "VSCode_latest_linux-x64.tar.gz";
        url = "https://vscode-update.azurewebsites.net/latest/linux-x64/insider";
      };
    }
  );
```

Reply:
how do you use vscode-with-extensions with vscode-insiders? When I do it like this:

```
{ pkgs ? import <nixpkgs> { config.allowUnfree = true; } }:

let
  vscode-insiders = ((pkgs.vscode.override {
      isInsiders = true;
    }).overrideAttrs (
      _: rec {
        name = "${pname}-${version}";
        pname = "vscode-insiders";
        version = "nightly";

        src = builtins.fetchurl {
          name = "VSCode_latest_linux-x64.tar.gz";
          url = "https://vscode-update.azurewebsites.net/latest/linux-x64/insider";
        };
      }
    ));
  extensions = [
    pkgs.vscode-extensions.bbenoist.Nix
  ];
  vscode-insiders-with-extensions = pkgs.vscode-with-extensions.override {
    vscode = vscode-insiders;
    vscodeExtensions = extensions;
  };
in
pkgs.mkShell {
  buildInputs = [
    vscode-insiders-with-extensions
  ];
}
```

Then the extensions are broken:

```
[nix-shell:~/nixpkgs-vscode]$ cat `which code-insiders`
#! /nix/store/j78ycszn9vgn0khl0lcpbd087x2mmr66-bash-4.4-p23/bin/bash -e
exec "/nix/store/ypi2jh6mzy9v5qq46w0pc3f0m9fjjg8g-vscode-insiders-nightly/bin/code-insiders"  --extensions-dir /nix/store/x8xnadwgpvprs0glp0m9nqyzxzm17m6v-vscode-insiders-extensions-nightly/share/vscode-insiders/extensions "$@"

[nix-shell:~/nixpkgs-vscode]$ ls /nix/store/x8xnadwgpvprs0glp0m9nqyzxzm17m6v-vscode-insiders-extensions-nightly/share/vscode-insiders/extensions
ls: cannot access '/nix/store/x8xnadwgpvprs0glp0m9nqyzxzm17m6v-vscode-insiders-extensions-nightly/share/vscode-insiders/extensions': No such file or directory

[nix-shell:~/nixpkgs-vscode]$ ls /nix/store/x8xnadwgpvprs0glp0m9nqyzxzm17m6v-vscode-insiders-extensions-nightly/share/vscode/extensions
bbenoist.Nix
```

So it seems this PR fixes that use case, rather than breaking it? Or is it common to use special 'insiders builds' of the extensions as well when running the vscode insiders build? Can you point me to an example of such an extension?

Reply:
When overriding, based on the order of overrides, they propagate throughout the package set, so
```nix
self: super:

{
  vscode = (
    super.vscode.override {
      isInsiders = true;
    }
  ).overrideAttrs (
    _: rec {
      name = "${pname}-${version}";
      pname = "vscode-insiders";
      version = "nightly";

      src = builtins.fetchurl {
        name = "VSCode_latest_linux-x64.tar.gz";
        url = "https://vscode-update.azurewebsites.net/latest/linux-x64/insider";
      };
    }
  );

  vscode-with-extensions = super.vscode-with-extensions.override {
    vscodeExtensions = with super.vscode-extensions; [
      ms-python.python
      ms-vscode.cpptools
      WakaTime.vscode-wakatime
    ] ++ super.vscode-utils.extensionsFromVscodeMarketplace [
       ...
``` 

since `vscode` was overrided before `vscode-with-extensions` the dependency on `vscode` in `vscode-with-extensions` points to the overriden version.

Here's how the files look for me (without this PR)
```
nixpkgs on  develop [$] 
➜ nix build nixpkgs.vscode-with-extensions 

nixpkgs on  develop [$] took 4s 
➜ cat result/bin/code-insiders 
#! /nix/store/j78ycszn9vgn0khl0lcpbd087x2mmr66-bash-4.4-p23/bin/bash -e
exec "/nix/store/ypi2jh6mzy9v5qq46w0pc3f0m9fjjg8g-vscode-insiders-nightly/bin/code-insiders"  --extensions-dir /nix/store/z29gg27qiy42v9ps22mfr5pb1jm5fq69-vscode-insiders-extensions-nightly/share/vscode-insiders/extensions "$@"

nixpkgs on  develop [$] 
➜ l /nix/store/z29gg27qiy42v9ps22mfr5pb1jm5fq69-vscode-insiders-extensions-nightly/share/  
total 0
dr-xr-xr-x 3 root root 0 Dec 31  1969 .
dr-xr-xr-x 3 root root 0 Dec 31  1969 ..
dr-xr-xr-x 3 root root 0 Dec 31  1969 vscode-insiders
```

Reply:
Aah, gotcha. Pinning the directory it at the `vscodeExtension` side as well as you suggested.

Tested, works as expected with both variations.